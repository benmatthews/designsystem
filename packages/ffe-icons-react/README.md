# @sb1/ffe-icons-react

## Install

```
npm install --save @sb1/ffe-icons-react
```
