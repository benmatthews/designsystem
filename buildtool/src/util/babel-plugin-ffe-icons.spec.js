const babel = require('@babel/core');
const plugin = require('./babel-plugin-ffe-icons');

const actual = `import React from 'react';
import { AtvIkon, BamseIkon as MinBamseIkon } from '@sb1/ffe-icons-react';
const foo = 'bar';`;

describe('change to default imports', () => {
    const { code } = babel.transform(actual, { plugins: [plugin] });
    test('imports from lib path', () =>
        expect(code).toMatch('@sb1/ffe-icons-react/lib/bamse-ikon'));
    test('imports icon name', () =>
        expect(code).toMatch('import MinBamseIkon'));
    test('snapshot', () => expect(code).toMatchSnapshot());
});

describe('change to default imports from esm', () => {
    const { code } = babel.transform(actual, {
        plugins: [[plugin, { sourceSubpath: 'es' }]],
    });
    test('imports from es path', () =>
        expect(code).toMatch('@sb1/ffe-icons-react/es/bamse-ikon'));
    test('snapshot', () => expect(code).toMatchSnapshot());
});
