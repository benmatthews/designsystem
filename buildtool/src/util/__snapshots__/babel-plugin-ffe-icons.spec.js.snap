// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change to default imports from esm snapshot 1`] = `
"import React from 'react';
import AtvIkon from \\"@sb1/ffe-icons-react/es/atv-ikon\\";
import MinBamseIkon from \\"@sb1/ffe-icons-react/es/bamse-ikon\\";
const foo = 'bar';"
`;

exports[`change to default imports snapshot 1`] = `
"import React from 'react';
import AtvIkon from \\"@sb1/ffe-icons-react/lib/atv-ikon\\";
import MinBamseIkon from \\"@sb1/ffe-icons-react/lib/bamse-ikon\\";
const foo = 'bar';"
`;
